### ISSUE TITLE
Function libxml_disable_entity_loader() is deprecated - PHP 8.0 compatibility

### ISSUE THREAD

**Issue by danutavadanei** [SAML-Toolkits/php-saml#453]
Looks like this library is not compatible with PHP 8.0.

https://php.watch/versions/8.0/libxml_disable_entity_loader-deprecation is deprecated.

Any plan to add PHP 8.0 compatibility?

**Comment by pitbulk** [comment]
PHP 8.0 was officially released 1 week ago.

That said, we plan to release a compatible version of PHP 8.0

PRs al welcome to port the code, the use of
```
if (\PHP_VERSION_ID < 80000) {
    libxml_disable_entity_loader(true);
}
```
seems simple but for sure there are more incompatibilities like:
```
PHP 8.0 no longer allows calling class methods as a static method.
```
and the toolkit has some of them, for example:
```
Utils::extractOriginalQueryParam
Utils::calculateX509Fingerprint
Utils::castKey
Utils::getSelfURL
Utils::getSelfURLhost
Utils::getBaseURLPath
Utils::isSupportedSigningAlgorithm
Utils::getExpireTime
Utils::parseSAML2Time
Utils::parseTime2SAML
Utils::formatPrivateKey
Utils::formatCert
Utils::generateUniqueID
Utils::loadXML
Utils::query
Utils::validateXML
Utils::decryptElement

XMLSecEnc::staticLocateKeyInfo

Metadata::addX509KeyDescriptors
Metadata::builder
```

**Comment by danutavadanei** [comment]
I would go ahead and create a PR, I already have a working fork but I
really don't understand the branching model of this project. 🥺

If you can spare coupe of minutes to fill the gap I'll give a shot on
creating a PR.

**Comment by pitbulk** [comment]
master and 2.X branches contain the old php-saml with is only compatible with PHP5 and has no namespaces
3.X branches contain the version compatible with PHP 5 and PHP 7 and has namespaces

The first lines of the README contain that info

**Comment by fideloper** [comment]
@danutavadanei Do you happen to have a PR ready? I'd like to not duplicate work!

**Comment by danutavadanei** [comment]
@fideloper no. :( Couldn't find the time to work on this yet.

**Comment by fideloper** [comment]
Right on, i’ll see what I can do!

**Comment by pitbulk** [comment]
@arturslogins the [3.6.0](https://github.com/onelogin/php-saml/commits/3.6.0) branch I believe is compatible with PHP 8.0

**Comment by arturslogins** [comment]
@pitbulk  Thanks will try it out

**Comment by arturslogins** [comment]
@pitbulk  When you plan to release master?

**Comment by arturslogins** [comment]
Problem 1
    - onelogin/php-saml 3.6.0.x-dev requires phpunit/phpunit <7.5.18 -> found phpunit/phpunit[3.7.0, ..., 3.7.38, 4.0.0, ..., 4.8.36, 5.0.0, ..., 5.7.27, 6.0.0, ..., 6.5.14, 7.0.0, ..., 7.5.17] but it confl
icts with your root composer.json require (^9.3).

**Comment by pitbulk** [comment]
You have installed previously phpunit 9.3, uninstall it and try again

**Comment by arturslogins** [comment]
But why you included old phpunit lib version ?

**Comment by pitbulk** [comment]
phpunit greater than 7.5 17 forces PHP7 and I want to keep compatibility with PHP5.6

It is a library used for development so it does not affect the php-saml library at all

**Comment by arturslogins** [comment]
> phpunit greater than 7.5 17 forces PHP7 and I want to keep compatibility with PHP5.6
> 
> phpunit is used for development so it does not affect your project using php-saml library at all



So basically, I should use now in my project deprecated PHPunit version (https://phpunit.de/supported-versions.html) If I want to install 3.6.0.  branch ?

**Comment by pitbulk** [comment]
In the short term yes.

I'm considering the release of a 4.X branch that will force php > 7.2 and compatible with PHP 8 and new phpunit, and we will to port the unit test as well.

**Comment by arturslogins** [comment]
> In the short term yes.
> 
> I'm considering the release of a 4.X branch that will force php > 7.2 and compatible with PHP 8 and new phpunit, and we will to port the unit test as well.

When you plan to release it?

**Comment by pitbulk** [comment]
I don't have an ETA yet sorry.

I need first to work on: 
https://github.com/onelogin/php-saml/issues/457

as well as review the work done by @fideloper at
https://github.com/onelogin/php-saml/pull/459

**Comment by andrevalentin** [comment]
Following this one, from what I can see the 4.X branch will be the best solution here to get PHP8 support fully, both with the already merged change into 3.6.0 as well as the changes with phpunit. Looking forward to seeing when this will be a reality.
